ambry-mysql/src/main/java/com/github/ambry/repair/MysqlRepairRequestsDb.java

```diff
@@ -205,6 +221,57 @@ public List<RepairRequestRecord> getRepairRequests(long partitionId) throws SQLE
     }
   }
 
+  /**
+   * Select the records from one partition but exclude the record with this source name and port.
+   * @param partitionId partition id
+   * @param sourceHostName the host name of the source replica
+   * @param sourceHostPort the host port of the source replica
+   * @return the oldest {@link RepairRequestRecord}s.
+   */
+  @Override
+  public List<RepairRequestRecord> getRepairRequests(long partitionId, String sourceHostName, int sourceHostPort)
+      throws SQLException {
+    // private static final String GET_QUERY_EXCLUDE_SOURCE_REPLICA = String.format(""
+    //    + "SELECT %s, %s, %s, %s, %s, %s, %s, %s "
+    //    + "FROM %s "
+    //    + "WHERE %s = ? and (%s != ? or %s != ?) "
+    //    + "ORDER BY %s ASC "
+    //    + "LIMIT ?",
+    //    BLOB_ID, PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT, OPERATION_TYPE, OPERATION_TIME, LIFE_VERSION, EXPIRATION_TYPE,
+    //    REPAIR_REQUESTS_TABLE,
+    //    PARTITION_ID, SOURCE_HOST_NAME, SOURCE_HOST_PORT,
+    //    OPERATION_TIME);
+    try (Connection connection = dataSource.getConnection()) {
+      try (PreparedStatement statement = connection.prepareStatement(GET_QUERY_EXCLUDE_SOURCE_REPLICA)) {
+        statement.setLong(1, partitionId);
+        statement.setString(2, sourceHostName);
+        statement.setInt(3, sourceHostPort);
+        statement.setInt(4, config.listMaxResults);
+        try (ResultSet resultSet = statement.executeQuery()) {
+          List<RepairRequestRecord> result = new ArrayList<>();
+          while (resultSet.next()) {
+            String blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(1));
+            // resultSet.getLong(2) is the partition id.
```

Comment:
nit: if you don't need grep partition id, you can just remove from the query. unless some other method need to use this query to get partition id.

Reply:
Remove the partition from the query. Thank you.